src/encoded/tests/test_types_experiment.py

```diff
@@ -12,80 +12,173 @@ def base_experiment(testapp, lab, award, cell_free):
     }
     return testapp.post_json('/experiment', item, status=201).json['@graph'][0]
 
-def test_isogenic_replicate_type(testapp, base_experiment, donor_1, donor_2,biosample_1, biosample_2, library_1, library_2, replicate_1_1, replicate_2_1 ):
+
+def create_fly_experiment(
+    testapp,
+    base_experiment,
+    fly, 
+    whole_organism, 
+    fly_donor, 
+    biosample_1, 
+    biosample_2, 
+    library_1, 
+    library_2, 
+    replicate_1_1, 
+    replicate_2_1, 
+    fastq_file_1, 
+    fastq_file_2,
+    ):
+    testapp.patch_json(fastq_file_1['@id'], {'dataset': base_experiment['@id'], 'replicate': replicate_1_1['@id']})
+    testapp.patch_json(fastq_file_2['@id'], {'dataset': base_experiment['@id'], 'replicate': replicate_2_1['@id']})
+    testapp.patch_json(biosample_1['@id'], {'donor': fly_donor['@id'], 'organism': fly['@id'], 'biosample_ontology': whole_organism['@id']})
+    testapp.patch_json(biosample_2['@id'], {'donor': fly_donor['@id'], 'organism': fly['@id'], 'biosample_ontology': whole_organism['@id']})
+    testapp.patch_json(library_1['@id'], {'biosample': biosample_1['@id']})
+    testapp.patch_json(library_2['@id'], {'biosample': biosample_2['@id']})
+    testapp.patch_json(replicate_1_1['@id'], {'library': library_1['@id'], 'experiment': base_experiment['@id']})
+    testapp.patch_json(replicate_2_1['@id'], {'library': library_2['@id'], 'experiment': base_experiment['@id']})
+   
+
+def test_fly_isogenic_replicate_type(
+    testapp,
+    base_experiment,
+    fly, 
+    whole_organism, 
+    fly_donor, 
+    biosample_1, 
+    biosample_2, 
+    library_1, 
+    library_2, 
+    replicate_1_1, 
+    replicate_2_1, 
+    fastq_file_1, 
+    fastq_file_2,
+    ):
+    create_fly_experiment(
+        testapp,
+        base_experiment,
+        fly, 
+        whole_organism, 
+        fly_donor, 
+        biosample_1, 
+        biosample_2, 
+        library_1, 
+        library_2, 
+        replicate_1_1, 
+        replicate_2_1, 
+        fastq_file_1, 
+        fastq_file_2,
+    )
+    res = testapp.get(base_experiment['@id']+'@@index-data') 
+    assert res.json['object']['replication_type']=='isogenic' 
+
+def test_fly_unreplicated_replicate_type(testapp, base_experiment, fly, whole_organism, fly_donor, biosample_1, biosample_2, library_1, library_2, replicate_1_1, replicate_2_1, fastq_file_1, fastq_file_2):
+    create_fly_experiment(
+        testapp,
+        base_experiment,
+        fly, 
+        whole_organism, 
+        fly_donor, 
+        biosample_1, 
+        biosample_2, 
+        library_1, 
+        library_2, 
+        replicate_1_1, 
+        replicate_2_1, 
+        fastq_file_1, 
+        fastq_file_2,
+    )
+    testapp.patch_json(replicate_2_1['@id'], {'library': library_2['@id'], 'experiment': base_experiment['@id'], 'status': 'deleted'})
+    res = testapp.get(base_experiment['@id']+'@@index-data') 
+    assert res.json['object']['replication_type']=='unreplicated' 
```

Comment:
If the `create_fly_experiment` is made into a fixture, then this can be simplified a lot too:
```suggestion
def test_fly_unreplicated_replicate_type(testapp, create_fly_experiment, library_2, replicate_2_1):
    testapp.patch_json(replicate_2_1['@id'], {'library': library_2['@id'], 'experiment': create_fly_experiment['@id'], 'status': 'deleted'})
    res = testapp.get(create_fly_experiment['@id']+'@@index-data') 
    assert res.json['object']['replication_type']=='unreplicated' 
```

Reply:
Done